src/main/java/org/squiddev/plethora/integration/astralsorcery/MethodsAstralSorcery.java

```diff
@@ -0,0 +1,68 @@
+package org.squiddev.plethora.integration.astralsorcery;
+
+import hellfirepvp.astralsorcery.AstralSorcery;
+import hellfirepvp.astralsorcery.common.auxiliary.CelestialGatewaySystem;
+import hellfirepvp.astralsorcery.common.data.world.WorldCacheManager;
+import hellfirepvp.astralsorcery.common.data.world.data.GatewayCache;
+import hellfirepvp.astralsorcery.common.tile.TileCelestialGateway;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.common.DimensionManager;
+import net.minecraftforge.fml.relauncher.Side;
+import org.squiddev.plethora.api.IWorldLocation;
+import org.squiddev.plethora.api.method.ContextKeys;
+import org.squiddev.plethora.api.method.IContext;
+import org.squiddev.plethora.api.method.LuaList;
+import org.squiddev.plethora.api.method.wrapper.PlethoraMethod;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class MethodsAstralSorcery {
+	private MethodsAstralSorcery() {
+	}
+
+	//REFINE This method is potentially OP, as it exposes all available `GatewayNode`s.
+	// That said, players can already name a gateway, and they could add signs to the destination
+	// so that it shows in the preview (if it is working; haven't tested)
+	// If Hellfire complains, it's easy enough to delete...
+	@PlethoraMethod(
+		modId = AstralSorcery.MODID,
+		doc = "-- Get a list of all Celestial Gateways, grouped by dimension"
+	)
+	public static Map<String, Object> getGateways(@Nonnull IContext<TileCelestialGateway> context) {
+		//REFINE Should be able to convert this to an annotated method arg...
+		IWorldLocation location = context.getContext(ContextKeys.ORIGIN, IWorldLocation.class);
+		if (location == null) return Collections.singletonMap("error", "Failed to get list of gateways, world data not in context.");
+
+		//TODO Confirm that this code does, in fact, only run on the server side... don't want to break things!
+		Map<Integer, List<GatewayCache.GatewayNode>> nodesByDimension = CelestialGatewaySystem.instance.getGatewayCache(Side.SERVER);
+		Map<String, Object> fullOut = new HashMap<>();
+
+		//TODO This will break badly if dimensions aren't identified by number, e.g. 1.13, NEID, JEID, etc.
+		for (Integer id : nodesByDimension.keySet()) {
```

Comment:
Can this just be an `int` here? Can never remember when unboxing applies.

Reply:
Honestly, not sure about implicit boxing/unboxing, but I've applied IDEA's suggestion to iterate the `entrySet` instead, so this no longer applies.